eclair-core/src/main/scala/fr/acinq/eclair/channel/Commitments.scala

```diff
@@ -183,8 +182,7 @@ object Commitments {
     val incomingHtlcs = reduced.htlcs.filter(_.direction == IN)
 
     val htlcValueInFlight = incomingHtlcs.map(_.add.amountMsat).sum
-    // we must use unsigned comparison here because 'maxHtlcValueInFlightMsat' is effectively an uint64 and can exceed the capacity of MilliSatoshi class
-    if (htlcValueInFlight.compareUnsigned(commitments1.localParams.maxHtlcValueInFlightMsat) > 0) {
+    if (commitments1.localParams.maxHtlcValueInFlightMsat < htlcValueInFlight) {
```

Comment:
The removed comment would probably have helped somewhere around line 30 in UInt64.scala.